### PR TITLE
ImageMagick7: Update to 7.1.1-29

### PR DESCRIPTION
**Pull request by Dave-Allured**
#### Description

* Update 7.1.1-28 --> 7.1.1-29
* Add legacy support portgroup.
* Maybe fixes 10.6 builds. We will find out on 10.6 builders after merge.
* In description, add pointer to legacy ImageMagick (version 6) port.
* Clean up obsolete comments.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.6.4 22G513 x86_64
Xcode 15.2 / Command Line Tools 15.1.0.

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tried a full install with `sudo port -vs install`?
- [ ] tested basic functionality of all binary files?
- [x] tested basic functionality of ONE binary file?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: graphics/ImageMagick7/Portfile [hunk 1]
```diff
@@ -3,6 +3,7 @@
 PortSystem          1.0
 PortGroup           conflicts_build 1.0
 PortGroup           github 1.0
+PortGroup           legacysupport 1.1
```
Comment (barracuda156) [inline graphics/ImageMagick7/Portfile:6]: While it is not required, but if we know what exactly fails without `legacysupport`, perhaps worth setting `legacysupport.newest_darwin_requires_legacy`.

Comment (Dave-Allured) [reply graphics/ImageMagick7/Portfile:6]: > if we know what exactly fails ...

Okay.  I threw in PG `legacysupport` on a total guess, to try to fix 10.6 builds.  Exactly what fails is incomplete `TargetConditionals.h`, from the builder logs.  It is the same for both x386 and x86_64.  Specifically, all three of these symbols are missing:
```
MagickCore/delegate.c:86:7: error: 'TARGET_OS_IOS' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
  #if TARGET_OS_IOS || TARGET_OS_WATCH || TARGET_OS_TV
  etc.
```
I do not know whether plain `legacysupport` will fix this.  I would be delighted if you can suggest a better approach, or a more targeted fix.

Comment (barracuda156) [reply graphics/ImageMagick7/Portfile:6]: Max Version Needing Feature OSX10.10
https://github.com/macports/macports-legacy-support

Comment (Dave-Allured) [reply graphics/ImageMagick7/Portfile:6]: Yes, there it is, `TargetConditionals.h` with the desired missing definitions.  So, okay to include that `legacysupport` portgroup, the way I have it now?  If there is a way to target that specific header, please show me how.

Comment (barracuda156) [reply graphics/ImageMagick7/Portfile:6]: @Dave-Allured I just meant we could add `legacysupport.newest_darwin_requires_legacy 14`

Comment (Dave-Allured) [reply graphics/ImageMagick7/Portfile:6]: @barracuda156 We posted at almost exactly the same time.  Yes, based on the last builder results, this particular legacy fix is needed only up to 10.6.  Let's see how this one goes.

**Review by barracuda156 — APPROVED** [review]

**Pull request merged**